lib/posix-process/exit.c

```diff
@@ -304,12 +304,12 @@ __noreturn void exit(int status)
 	 * process."
 	 */
 	uk_syscall_e_exit_group(status);
-	UK_BUG(); /* noreturn */
+	UK_CRASH("exit_group syscall returned"); /* noreturn */
```

Comment:
This looks more like a workaround. From what I see the problem originates from the fact that we can't declare syscalls (`uk_sys_exit_group()` in this case) as `__noreturn`. So perhaps we should provide a way to declare syscalls as noreturn? CC: @mogasergiu

Also, generally speaking, as these checks are a common pattern in our codebase: If we make a call *from* a noreturn function *to* a noreturn function, is  a check really necessary? Is it possible for a noreturn function one day to accidentally return?